main.js

```diff
@@ -134,246 +78,271 @@ var getBlocks = () => {
+                                'share': 0
+                            });
+                        }
+                        if (allVotersEver.has(tx.recipientId)) {
+                            let thusFar = totalBalanceThusFar.get(tx.recipientId);
+                            thusFar -= parseInt(tx.amount);
+                            totalBalanceThusFar.set(tx.recipientId, thusFar);
+                            sortedForgedBlocks[idx + 1].voterBalances.set(tx.recipientId, {
+                                'balance': thusFar,
+                                'share': 0
+                            });
+                        }
+                        // Apply votes
+                        if (tx.rawasset.includes(`-${config.pKey}`)) {
+                            currentVoters.add(tx.senderId);
+                            sortedForgedBlocks[idx + 1].voterBalances.set(tx.senderId, {
+                                'balance': totalBalanceThusFar.get(tx.senderId),
+                                'share': 0
+                            });
+                        } else if (tx.rawasset.includes(`+${config.pKey}`)) {
+                            currentVoters.delete(tx.senderId);
+                            sortedForgedBlocks[idx + 1].voterBalances.set(tx.senderId, {
+                                'balance': 0,
+                                'share': 0
+                            });
+                        }
+                        return true;
+                    }
+                }
+            })
+        }
+
+        sortedForgedBlocks[0].voterBalances = new Map(curBals.map(vote => [vote.address, { // Populate the highest block
+            'balance': vote.balance,
+            'share': 0
+        }]));
+        txs.forEach((tx, idx) => {
+            sortTx(tx);
+        });
+        // Populate the last blocks we missed
+        sortedForgedBlocks.forEach((block, idx) => {
+            if (!block.voterBalances) {
+                block.voterBalances = new Map(sortedForgedBlocks[idx - 1].voterBalances);
+            }
+        })
+        currentVoters = new Set(voterAddrs);
+        let voterSum = 0;
+        latestForgedBlock.voterBalances.forEach((bal) => voterSum += bal[0])
+        console.log("Pool balance calculations complete");
+
+        resolve();
+    })
+};
+
+let getVoterWeight = () => {
+    console.log("Calculating voter weights for each block...");
+    let taxes = new BigNumber(0);
+    let totalPay = new BigNumber(0);
+    return new Promise((resolve, rej) => {
+        sortedForgedBlocks = sortedForgedBlocks.reverse();
+        sortedForgedBlocks.forEach((block, idx) => {
+            //console.log("block - " + parseInt(idx + 1) + " / " + sortedForgedBlocks.length + " | blockVoterBal.size: " + block.voterBalances.size);
+
+            block.voterBalances.forEach((balanceData, index) => {
+                if (balanceData.balance > (config.cap * 100000000)) {
```

Comment:
The conversion from arktoshi to ark should be done earlier to make this logic easier to read. Also, extract the math to a function that can be reused instead of writing `* 10000000` everywhere.